src/movies.js

```diff
@@ -1,16 +1,164 @@
 // Iteration 1: All directors? - Get the array of all directors.
+
+
+
+function getAllDirectors(moviesArr) {
+  
+    const directors = moviesArr.map(function(movie){
+  
+    return movie.director 
+    })
+return directors
+}
+
 // _Bonus_: It seems some of the directors had directed multiple movies so they will pop up multiple times in the array of directors. How could you "clean" a bit this array and make it unified (without duplicates)?
 
+
 // Iteration 2: Steven Spielberg. The best? - How many drama movies did STEVEN SPIELBERG direct?
 
+//create function that gets all movies by Spielberg that are a drama
+
+// this gets me all the Spielberg Movies:
+function getAllSpielberg (moviesArr) {
+
+    const spielbergmovie = moviesArr.map (function(movie){
+
+        if (movie.director === 'Steven Spielberg')
+        return movie.director = 1
+      
+      else 
+      return movie.director = 0
+    })
+return spielbergmovie
+}
+
+// get all Drama: but this doesn't work because it only takes movies that are explicitly from the  Drama genre
+
+function getAllDrama (moviesArr) {
+    const allDrama = moviesArr.map (function(movie){
+      if (movie.genre[''] === 'Drama')
+        return  1
+      else return 0
+    })
+    return allDrama
+  }
+
+
+//all the Spielberg Dramas
+
+function howManyMovies(moviesArray)  {
+    const DramaSpielberg = moviesArray.filter(function(movie){
+      return movie.director === 'Steven Spielberg' && movie.genre.includes('Drama')
+    })
+    return DramaSpielberg.length
+  };
+
+
+
+//first Try down here, forgot the length 
+
+function howManyMovies(moviesArray)  {
+    const DramaSpielberg = moviesArray.filter(function(movie){
+      return movie.director === 'Steven Spielberg' && movie.genre.includes('Drama')
+    })
+    return DramaSpielberg
+  };
+  
+
 // Iteration 3: All rates average - Get the average of all rates with 2 decimals
 
+// I need the sum of all rates and divide them by the amount of all rates
+
+// I need to access all the rates
+// I need to count the rates
+// I need to divide the sum of all rates by the amount of rates
+
+// accessing all the rates: (got an reference error telling me that element is not defined)
+
+/*
+function ratesAverage (someArr) {
+    const rates = []
+    someArr.forEach(element => console.log(element.rate))
+    return element.rate
+}
+*/
+
+function ratesAverage (someArr) {
+    const rates = []
+    someArr.forEach((el) => {
+      if (el.rate){
+        rates.push(el.rate)  
+      }
+    })
+  const sum = rates.reduce(function (acc, val) {
+
+     return acc + val / rates.length
+ },0)
+ return sum.toFixed(2)
+}
```

Comment:
Hey Jeff, thank you for all that feedback!! I really appreciate it! 
One question that comes up here is why you returned Number in the end. We didn't define Number anywhere so I don't really understand the last step. We could also just discuss this in class tomorrow if you prefer that instead of writing it here :)